toolchain/sem_ir/ids.h

```diff
@@ -301,6 +301,8 @@ struct GenericId : public IdBase<GenericId> {
 // The ID of a specific, which is the result of specifying the generic arguments
 // for a generic.
 struct SpecificId : public IdBase<SpecificId> {
+  using DiagnosticType = DiagnosticTypeInfo<std::string>;
```

Comment:
Is this needed? It's making an error.

```
In file included from toolchain/sem_ir/ids.cpp:5:
./toolchain/sem_ir/ids.h:304:26: error: no template named 'DiagnosticTypeInfo'
  using DiagnosticType = DiagnosticTypeInfo<std::string>;
                         ^
```

But I saw that we are handling it in toolchain/check/sem_ir_loc_diagnostic_emitter.cpp as well. I am actually not sure how the DiagnosticType gets consumed.

Reply:
This is a conflict with #5173 which renamed this to `Diagnostics::TypeInfo`, fixed. This typedef is detected by the diagnostics machinery, and is needed in order to allow using `SpecificId` as an argument type in diagnostics.